File: src/statusbar/modules/notes.ts
import type {
  StatusBarModule,
  StatusBarModuleContext,
  Segment,
} from "../types.ts";

export const notesModule: StatusBarModule = {
  id: "notes",
  render(ctx: StatusBarModuleContext): Segment[] {
    const count = ctx.metrics.notesCount;
    const icon = ctx.icons ? "󱓧 " : "";
    return [
      {
        text: ` ${icon}${count} `,
        fg: ctx.colors.fg,
        bg: ctx.colors.bg,
      },
    ];
  },
};

File: rust/crates/maxmux-statusbar/src/modules/notes.rs
use crate::modules::{ModuleContext, StatusBarModule};
use crate::types::Segment;

pub struct NotesModule;

impl StatusBarModule for NotesModule {
    fn id(&self) -> &str {
        "notes"
    }

    fn render(&self, ctx: &ModuleContext) -> Vec<Segment> {
        let count = ctx.metrics.notes_count;
        let text = if ctx.icons {
            format!(" \u{f1ce7} {} ", count)
        } else {
            format!(" {} ", count)
        };
        vec![Segment::new(&text, &ctx.colors.fg, &ctx.colors.bg)]
    }
}
